### PR TITLE
TSD-54: Cancelled Order Feed

### PR DESCRIPTION
**Pull request by ptrollinspt**
- Created a new jobstep for cancelled orders export
- Created a new steptype for cancelled orders
- Updated job and system metadata zip
- added a feedStyle arg to replace hardcode feedstyle
- Added new write helper function for cancelled orders
- Updated urls to https

### CODE REVIEW

File: cartridges/int_turnto_core_v5/cartridge/scripts/util/ServiceFactory.js [hunk 1]
```diff
@@ -153,18 +152,16 @@ var ServiceFactory = {
         arrayOfRequestParts.push(new HTTPRequestPart('file', file));
         arrayOfRequestParts.push(new HTTPRequestPart('siteKey', siteKey));
         arrayOfRequestParts.push(new HTTPRequestPart('authKey', authKey));
-        arrayOfRequestParts.push(new HTTPRequestPart('feedStyle', 'tab-style.1'));
+        arrayOfRequestParts.push(new HTTPRequestPart('feedStyle', feedStyle));
```
Comment (rbonsall-turnto) [inline cartridges/int_turnto_core_v5/cartridge/scripts/util/ServiceFactory.js:155]: Just for my own understanding.  How are these request parts used?  In other words, what is it calling?

File: cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/ExportCancelledOrders.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    if (!turntoDir.exists()) {
+                        turntoDir.mkdirs();
+                    }
+
+                    // Initialize a file writer for output
+                    var orderExportFile = new File(turntoDir.getFullPath() + File.SEPARATOR + exportFileName + '_' + currentLocale + '_' + Site.getCurrent().ID + '.txt');
+
+                    fileWriter = new FileWriter(orderExportFile);
+
+                    fileWriter.writeLine('ORDERID\tSKU');
```
Comment (rbonsall-turnto) [inline cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/ExportCancelledOrders.js:78]: This appears to be the header for the feed file.  It appears the `OrderWriterHelper` call below writes out a log more fields that just order id and sku.  How does this work?

Comment (ptrollinspt) [reply cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/ExportCancelledOrders.js:78]: I created a second method `writeCancelledOrderData` in `OrderWriterHelper` that only writes the two fields. The original method `writeOrderData` writes all the order fields.

File: cartridges/int_turnto_core_v5/cartridge/scripts/util/ServiceFactory.js (hunk 1 above)
Comment (ptrollinspt) [reply cartridges/int_turnto_core_v5/cartridge/scripts/util/ServiceFactory.js:155]: This logic was already in place but the `requestDataContainer` is returned with the parts as args. This is passed to FeedUploadService and used in [client.sendMultiPart(requestObject.args)](https://github.com/turnto/salesforce-commerce-cloud/blob/master/cartridges/int_turnto_core_v5/cartridge/scripts/service/FeedUploadService.js#L32)

File: cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/ExportCancelledOrders.js (hunk 2 above)
Comment (rbonsall-turnto) [reply cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/ExportCancelledOrders.js:78]: Thanks for the clarification.  I was looking at the `OrderWriterHelper` on master, totally missing the one in the PR :facepalm:

File: cartridges/int_turnto_core_v5/cartridge/scripts/util/ServiceFactory.js (hunk 1 above)
Comment (rbonsall-turnto) [reply cartridges/int_turnto_core_v5/cartridge/scripts/util/ServiceFactory.js:155]: Got it, thanks.  Make more sense to me now.

**Review by rbonsall-turnto — APPROVED** [review]

**Pull request merged**